### DISCUSSION TITLE
Installation on Windows 10

### DISCUSSION THREAD

**Discussion by HEHenson** [somnathrakshit/geograpy3#67] [Q&A] [answered]
I am working with Python 3.10 and I am unable to install geograpy3 into a venv although I can do it on Linux without any problems.  I have Microsoft C++ on my machine but it still cannot find lxml when I do pip in the venv.  In fact I can even install outside of the venv.

I am sure that this is a temporary problem but in the meantime, any help would be greatly appreciated.

**Comment by somnathrakshit** [comment]
Can you share the command used and the output?

**Reply by HEHenson** [reply]
I actually found a solution.  I just installed the missing lmxl from https://www.lfd.uci.edu/~gohlke/pythonlibs/#lxml .  I know it is a patch, or a short-term solution but it worked.

**Comment by WolfgangFahl** [comment, marked as answer]
@HEHenson i had problems with lxml recently myself see https://bugs.launchpad.net/lxml/+bug/1997267. If the library is just used and not built you should be fine though. "3.10" and Windows is in the build matrix for the CI of this project so there should not be a problem in principle. (Except that in my experience Windows has been a nightmare for far too long now ...)

**Reply by HEHenson** [reply]
None of this should have been a problem in principle.  Alas, we live in an imperfect world.  I think I was lucky in finding the solution that I mention above.

**Comment by somnathrakshit** [comment]
@HEHenson @WolfgangFahl I have been thinking today on how to fix this problem. One potential way is to install a package called pipwin, which automatically downloads and installs python packages from https://www.lfd.uci.edu/~gohlke/pythonlibs repository.

A downside to this solution is that we are adding an extra python package (pipwin). An ideal solution would be to install pipwin only on Windows and skip its installation for macOS and Linux. 

Let me know what you think.

**Reply by HEHenson** [reply]
If you are willing to do that it would be a big help.  I don't know if there would be any legal problems.  However, I would be glad to help out on the testing and documentation side.

**Comment by WolfgangFahl** [comment]
@somnathrakshit  - Hi Somnath - nice to hear from your again. While i understand your urge to help the poor Windows users i think that the problem(s) need to be solved by the folks in Redmond, WA (or by the users for that matter - see [Ubuntu bug #1](https://bugs.launchpad.net/ubuntu/+bug/1)
